### PR TITLE
Test of Register and Signin.

### PR DESCRIPTION
**Pull request by TTL2000**
I add the test of after Register and Signin return to homepage.

### CODE REVIEW

**Review by RileyYe — APPROVED** [review]
approved

**Comment by PeterlitsZo** [comment]
> approved

Did you review really?

**Comment by PeterlitsZo** [comment]
Do you really run those tests? Why did they fail?

**Comment by PeterlitsZo** [comment]
Hello, @TTL2000, here are some suggestions for you:

1. Commit with a good message style. You can click [here](https://github.com/ACMHomepage/ACMHomepage/blob/master/docs/contribute.md) to know more. If you have some questions, please feel free to open an issue to talk about it.
2. If you work on your repository (TTL2000/ACMHomepage) rather than this repository, feel free to use `git rebase` and `git push --force`.
3. Use another branch, for example, `add-test-for-register-and-signin` rather than **`master`**!

File: front/src/page/Register.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    </ApolloProvider>
+  </Provider>
+);
+const server = setupServer(registerHandler);
+
+beforeAll(() => server.listen());
+
+beforeEach(() => server.resetHandlers());
+
+afterAll(() => server.close());
```
Comment (PeterlitsZo) [inline front/src/page/Register.test.tsx:30]: I think putting those lines into the `jest.setup.ts` will be a better idea.

File: front/src/page/Register.test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <Routes>
+          <Route path="/" element={<Homepage />} />
+          <Route path="register" element={<Resgister />} />
+          <Route path="*" element={<_404 />} />
+        </Routes>
+      </MemoryRouter>
+    </RegisterWapper>,
+  );
+
+  expect(screen.queryByText('__Homepage__')).toBeNull();
```
Comment (PeterlitsZo) [inline front/src/page/Register.test.tsx:64]: Add comments please.

File: front/src/api/auth.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+      } catch (_err) {
+        /* Do nothing, because it will be catched by redux store. */
+      }
+    },
+    [signIn],
+  );
+};
+
+/******************************************************************************
+ * Sign up part
```
Comment (PeterlitsZo) [inline front/src/api/auth.ts:107]: Rename to `Register part`

**Review by PeterlitsZo — CHANGES_REQUESTED** [review]

**Review by TTL2000 — COMMENTED** [review]
OK，and i'd like to know why putting some lines of Register.test.tsx to jest.setup.ts is better!

**Pull request merged**